accounts-db/src/accounts_db.rs

```diff
@@ -3904,23 +3904,24 @@ impl AccountsDb {
 
     /// sort `accounts` by pubkey.
     /// Remove earlier entries with the same pubkey as later entries.
+    #[cfg_attr(feature = "dev-context-only-utils", qualifiers(pub))]
     fn sort_and_remove_dups(accounts: &mut Vec<AccountFromStorage>) {
         // stable sort because we want the most recent only
         accounts.sort_by(|a, b| a.pubkey().cmp(b.pubkey()));
         if accounts.len() > 1 {
-            let mut i = 0;
-            // iterate 0..1 less than end
-            while i < accounts.len() - 1 {
-                let current = accounts[i];
-                let next = accounts[i + 1];
-                if current.pubkey() == next.pubkey() {
-                    // remove the first duplicate
-                    accounts.remove(i);
-                    // do not advance i, we just removed an element at i
-                    continue;
+            let mut last = 0;
```

Comment:
we should put some metrics on how many dups there are. I expect the # to be approx. 0. In that case, we are sort cost + O(n) to iterate once and make sure there are no dups.

Reply:
and this is not worth optimizing. I think this is cya.

Reply:
It seems that the new impl in this PR is faster than the old one. I think the if condition check in the old code is not as processor pipeline friendly as the new loop without the conditional check. 

```
old: test bench_sort_and_remove_dups_no_dups                ... bench:      55,573.88 ns/iter (+/- 2,410.57)
new: test bench_sort_and_remove_dups_no_dups                ... bench:      50,089.04 ns/iter (+/- 1,426.60)
```